controllers/k8ssandra/cleanup.go

```diff
@@ -478,3 +478,47 @@ func (r *K8ssandraClusterReconciler) deleteCronJobs(
 	}
 	return
 }
+
+func (r *K8ssandraClusterReconciler) deleteCassandraDatacenter(
+	ctx context.Context,
+	dcTemplate k8ssandraapi.CassandraDatacenterTemplate,
+	namespace string,
+	remoteClient client.Client,
+	logger logr.Logger,
+) (hasErrors bool, earlyReturn *result.ReconcileResult) {
+	dcKey := client.ObjectKey{Namespace: namespace, Name: dcTemplate.Meta.Name}
+	dc := &cassdcapi.CassandraDatacenter{}
+	err := remoteClient.Get(ctx, dcKey, dc)
+	if err != nil {
+		// If the DC is not found, it means it has already been deleted.
+		if !errors.IsNotFound(err) {
+			logger.Error(err, "Failed to get CassandraDatacenter for deletion",
+				"CassandraDatacenter", dcKey, "Context", dcTemplate.K8sContext)
+			hasErrors = true
+		}
+		return
+	}
+
+	// Remove finalizer before deletion
+	if controllerutil.ContainsFinalizer(dc, k8ssandra.K8ssandraClusterFinalizer) {
+		patch := client.MergeFrom(dc.DeepCopy())
+		controllerutil.RemoveFinalizer(dc, k8ssandra.K8ssandraClusterFinalizer)
+		if err := remoteClient.Patch(ctx, dc, patch); err != nil {
+			logger.Error(err, "Failed to remove finalizer from CassandraDatacenter", "CassandraDatacenter", dcKey, "Context", dcTemplate.K8sContext)
+			hasErrors = true
+			return
+		}
+	}
+
+	if err = remoteClient.Delete(ctx, dc); err != nil {
+		logger.Error(err, "Failed to delete CassandraDatacenter", "CassandraDatacenter", dcKey, "Context", dcTemplate.K8sContext)
+		hasErrors = true
+	} else {
+		// If the DC is found, it means it has not been deleted yet.
+		// We need to requeue to wait for the deletion to complete.
+		logger.Info("Waiting for actual deletion of CassandraDatacenter", "CassandraDatacenter", utils.GetKey(dc))
+		result := result.RequeueSoon(r.DefaultDelay)
+		earlyReturn = &result
```

Comment:
issue: And this is just wrong. Why is there a pointer to a pointer? I don't see a justification here.